examples/httpget/httpget.pony

```diff
@@ -3,58 +3,192 @@ use "collections"
 use "net/http"
 use "net/ssl"
 use "files"
+use "encode/base64"
+use "options"
 
 actor Main
-  let _env: Env
+  """
+  Fetch data from URLs on the command line.
+  """
+  var user: String = ""
+  var pass: String = ""
+  var output: String = ""
 
   new create(env: Env) =>
+    // Get common command line options.
+    let opts = Options(env.args)
+    opts
+      .add("user", "u", StringArgument)
+      .add("pass", "p", StringArgument)
+      .add("output", "o", StringArgument)
+
+    for option in opts do
+      match option
+       | ("user", let u: String) => user = u
+       | ("pass", let p: String) => pass = p
+       | ("output", let o: String) => output = o
+       | let err: ParseError =>
+           err.report(env.out)
+           _usage( env.out )
+           return
+       end
+      end
+
+    // The positional parameter is the URL to be fetched.
+    let urlref = try env.args( env.args.size() - 1 ) else "" end
+    let url = try URL.valid(urlref)
+      else
+        env.out.print("Invalid URL")
+        return
+      end
+
+    // Start the actor that does the real work.
+    _GetWork.create( env, url, user, pass, output )
+
+  fun tag _usage( out: StdStream ) =>
+    """
+    Print command error message.
+    """
+    out.print(
+      "httpget [OPTIONS] URL\n" +
+      "  --user (-u)  Username for authenticated queries\n" +
+      "  --pass (-p)  Password for authenticated queries\n" +
+      "  --output (-o) Name of file to write response body\n"
+      )
+
+actor _GetWork
+  """
+  Do the work of fetching a resource
+  """
+  let _env: Env
+
+  new create( env: Env, url: URL, user: String, pass: String,
+    output: String ) =>
```

Comment:
@jemc i did. man this PR, it makes my browser so slow when I'm on the review page and I type something and there is a second delay before anything shows up. I'm surprised there aren't more I Fubar'd.